lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -501,7 +513,31 @@ PlaylistGenerator.prototype.reset = function() {
     return  that.resetting;
 };
 
+// todo: set ended entry PresentationEndTime
+PlaylistGenerator.prototype.end = function() {
+
+    return persistPlayList.call(this, true, true)
+        .then(() => {
+            this.logger.debug(`@@@ [${this.entryId}], stream ended.`);
+        })
+        .catch((err) => {
+            this.logger.error(`Error ending stream of entry [${this.entryId}]: , error: ${ErrorUtils.error2string(err)}`);
+        });
+}
 
+// update presentationEndTime and save playlist
+PlaylistGenerator.prototype.persistStopped = function() {
+    return persistPlayList.call(this, true);
+}
 
+// update presentationEndTime and expireTime and save to disk
+PlaylistGenerator.prototype.restoreStopped = function() {
+    this.start();
```

Comment:
no need for this , since you will update presentationEndTime and expirationTime when updating the playlist